File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> prefixSum = new HashMap<>();
        int answer = 0, currentPrefix = 0;
        prefixSum.put(0,1);
        for(int i = 0; i < nums.length; i++){
            currentPrefix += nums[i];
            answer += prefixSum.getOrDefault(currentPrefix -k, 0);
            prefixSum.put(currentPrefix,prefixSum.getOrDefault(currentPrefix, 0) + 1);
        }
        return answer;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int, int> prefixSum;
        int answer = 0, currentPrefix = 0;
        prefixSum[0]++;
        for(int i = 0; i <nums.size(); i++)
        {
            currentPrefix += nums[i];
            answer += prefixSum[currentPrefix - k];
            prefixSum[currentPrefix]++;
        }
        return answer;
    }
};